policy/modules/system/systemd.te

```diff
@@ -767,6 +770,36 @@ init_search_runtime(systemd_hw_t)
 seutil_read_config(systemd_hw_t)
 seutil_read_file_contexts(systemd_hw_t)
 
+#######################################
+#
+# journald local policy
+#
+# During system boot, the service systemd-journal-catalog-update.service
+# runs journalctl with the switch --update-catalog which needs manage
+# permissions for systemd_journal_t files.  Transitioning from initrc_t
+# into systemd_journal_init_t for this operation limits write access
+# to sysemd_journal_t files to only the systemd_journal_init_t domain.
+#
+
+dontaudit systemd_journal_init_t self:capability net_admin;
+
+manage_files_pattern(systemd_journal_init_t, systemd_journal_t, systemd_journal_t)
+
+fs_getattr_cgroup(systemd_journal_init_t)
+fs_search_cgroup_dirs(systemd_journal_init_t)
+
+kernel_getattr_proc(systemd_journal_init_t)
+kernel_read_kernel_sysctls(systemd_journal_init_t)
+kernel_read_system_state(systemd_journal_init_t)
+
+init_read_state(systemd_journal_init_t)
+init_search_var_lib_dirs(systemd_journal_init_t)
+
+logging_send_syslog_msg(systemd_journal_init_t)
+logging_write_runtime_sockets(systemd_journal_init_t)
```

Comment:
It looks like this is the only socket operation but there's no explicit connectto/sendto other than in the line above. If this is part of connecting to journald, it would seem there is a wrong domain somewhere?

Reply:
This is the error I'm seeing when this isn't being allowed:
Sep 23 02:56:37 localhost audisp-syslog[1745]: node=localhost type=AVC msg=audit(1695437796.749:131): avc:  denied  { write } for  pid=1707 comm="journalctl" name="io.systemd.journal" dev="tmpfs" ino=723 scontext=system_u:system_r:systemd_journal_init_t:s0 tcontext=system_u:object_r:syslogd_runtime_t:s0 tclass=sock_file permissive=1

Sep 23 02:56:37 localhost audisp-syslog[1745]: node=localhost type=SYSCALL msg=audit(1695437796.749:131): arch=c000003e syscall=42 success=yes exit=0 a0=4 a1=7ffefe90c610 a2=2a a3=5593de6631f0 items=0 ppid=1 pid=1707 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="journalctl" exe="/usr/bin/journalctl" subj=system_u:system_r:systemd_journal_init_t:s0 key=(null) ARCH=x86_64 SYSCALL=connect AUID="unset" UID="root" GID="root" EUID="root" SUID="root" FSUID="root" EGID="root" SGID="root" FSGID="root"

Reply:
Got some help from a systemd maintainer. This is the varlink socket,so I'd prefer a single interface that handles this named something like `logging_stream_connect_journald_varlink`

Reply:
Maybe I did what you are asking.  I Just renamed the previously created interface to your suggested name, but now, re-reading your comment, I'm wondering if you wanted more in this new interface that might include some of what is in logging_send_syslog_msg.  
Let me know if you want more changes.

Reply:
The interface implementation should be effectively be `stream_connect_pattern`, not just the socket write. i.e. it should work without the `logging_sen_syslog_msg` call.

Reply:
Ok, I have updated the logging_stream_connect_journald_varlink interface as described.   But, I also left in the use of logging_send_syslog_msg because after altering the new interface to how it is now, I was still seeing the following denials.  I can add more rules to allow these denials into logging_stream_connect_journald_varlink, but it seems pretty redundant with the use of logging_send_syslog_msg.  If you want changes, just let me know.

Sep 26 16:39:07 localhost audisp-syslog[1552]: node=localhost type=AVC msg=audit(1695746346.769:132): avc:  denied  { create } for  pid=1525 comm="journalctl" scontext=system_u:system_r:systemd_journal_init_t:s0 tcontext=system_u:system_r:systemd_journal_init_t:s0 tclass=unix_dgram_socket permissive=1
Sep 26 16:39:07 localhost audisp-syslog[1552]: node=localhost type=AVC msg=audit(1695746346.769:133): avc:  denied  { getopt } for  pid=1525 comm="journalctl" scontext=system_u:system_r:systemd_journal_init_t:s0 tcontext=system_u:system_r:systemd_journal_init_t:s0 tclass=unix_dgram_socket permissive=1
Sep 26 16:39:07 localhost audisp-syslog[1552]: node=localhost type=AVC msg=audit(1695746346.769:134): avc:  denied  { setopt } for  pid=1525 comm="journalctl" scontext=system_u:system_r:systemd_journal_init_t:s0 tcontext=system_u:system_r:systemd_journal_init_t:s0 tclass=unix_dgram_socket permissive=1
Sep 26 16:39:07 localhost audisp-syslog[1552]: node=localhost type=AVC msg=audit(1695746346.769:135): avc:  denied  { connect } for  pid=1525 comm="journalctl" scontext=system_u:system_r:systemd_journal_init_t:s0 tcontext=system_u:system_r:systemd_journal_init_t:s0 tclass=unix_dgram_socket permissive=1
Sep 26 16:39:07 localhost audisp-syslog[1552]: node=localhost type=AVC msg=audit(1695746346.769:135): avc:  denied  { write } for  pid=1525 comm="journalctl" name="socket" dev="tmpfs" ino=58 scontext=system_u:system_r:systemd_journal_init_t:s0 tcontext=system_u:object_r:devlog_t:s0 tclass=sock_file permissive=1
Sep 26 16:39:07 localhost audisp-syslog[1552]: node=localhost type=AVC msg=audit(1695746346.769:135): avc:  denied  { sendto } for  pid=1525 comm="journalctl" path="/run/systemd/journal/socket" scontext=system_u:system_r:systemd_journal_init_t:s0 tcontext=system_u:system_r:kernel_t:s0 tclass=unix_dgram_socket permissive=1
Sep 26 16:39:07 localhost audisp-syslog[1552]: node=localhost type=AVC msg=audit(1695746346.770:136): avc:  denied  { create } for  pid=1525 comm="journalctl" scontext=system_u:system_r:systemd_journal_init_t:s0 tcontext=system_u:system_r:systemd_journal_init_t:s0 tclass=unix_stream_socket permissive=1
Sep 26 16:39:07 localhost audisp-syslog[1552]: node=localhost type=AVC msg=audit(1695746346.770:137): avc:  denied  { connect } for  pid=1525 comm="journalctl" scontext=system_u:system_r:systemd_journal_init_t:s0 tcontext=system_u:system_r:systemd_journal_init_t:s0 tclass=unix_stream_socket permissive=1
Sep 26 16:39:07 localhost audisp-syslog[1552]: node=localhost type=AVC msg=audit(1695746346.775:138): avc:  denied  { write } for  pid=1525 comm="journalctl" scontext=system_u:system_r:systemd_journal_init_t:s0 tcontext=system_u:system_r:systemd_journal_init_t:s0 tclass=unix_stream_socket permissive=1
Sep 26 16:39:07 localhost audisp-syslog[1552]: node=localhost type=AVC msg=audit(1695746346.775:139): avc:  denied  { read } for  pid=1525 comm="journalctl" scontext=system_u:system_r:systemd_journal_init_t:s0 tcontext=system_u:system_r:systemd_journal_init_t:s0 tclass=unix_stream_socket permissive=1
Sep 26 16:39:07 localhost audisp-syslog[1552]: node=localhost type=AVC msg=audit(1695746347.294:143): avc:  denied  { create } for  pid=1535 comm="journalctl" scontext=system_u:system_r:systemd_journal_init_t:s0 tcontext=system_u:system_r:systemd_journal_init_t:s0 tclass=unix_dgram_socket permissive=1
Sep 26 16:39:07 localhost audisp-syslog[1552]: node=localhost type=AVC msg=audit(1695746347.297:144): avc:  denied  { getopt } for  pid=1535 comm="journalctl" scontext=system_u:system_r:systemd_journal_init_t:s0 tcontext=system_u:system_r:systemd_journal_init_t:s0 tclass=unix_dgram_socket permissive=1
Sep 26 16:39:07 localhost audisp-syslog[1552]: node=localhost type=AVC msg=audit(1695746347.297:145): avc:  denied  { setopt } for  pid=1535 comm="journalctl" scontext=system_u:system_r:systemd_journal_init_t:s0 tcontext=system_u:system_r:systemd_journal_init_t:s0 tclass=unix_dgram_socket permissive=1
Sep 26 16:39:07 localhost audisp-syslog[1552]: node=localhost type=AVC msg=audit(1695746347.297:146): avc:  denied  { connect } for  pid=1535 comm="journalctl" scontext=system_u:system_r:systemd_journal_init_t:s0 tcontext=system_u:system_r:systemd_journal_init_t:s0 tclass=unix_dgram_socket permissive=1
Sep 26 16:39:07 localhost audisp-syslog[1552]: node=localhost type=AVC msg=audit(1695746347.297:146): avc:  denied  { write } for  pid=1535 comm="journalctl" name="socket" dev="tmpfs" ino=58 scontext=system_u:system_r:systemd_journal_init_t:s0 tcontext=system_u:object_r:devlog_t:s0 tclass=sock_file permissive=1
Sep 26 16:39:07 localhost audisp-syslog[1552]: node=localhost type=AVC msg=audit(1695746347.297:146): avc:  denied  { sendto } for  pid=1535 comm="journalctl" path="/run/systemd/journal/socket" scontext=system_u:system_r:systemd_journal_init_t:s0 tcontext=system_u:system_r:kernel_t:s0 tclass=unix_dgram_socket permissive=1